sdks/python/apache_beam/io/gcsio.py

```diff
@@ -555,11 +596,15 @@ def _get_segment(self, start, size):
     """Get the given segment of the current GCS file."""
     if size == 0:
       return ''
+    # The objects self.downloader and self.download_stream may be recreated if
```

Comment:
Is it possible that the recreation happens between the two 

```
downloader = self.downloader
download_stream = self.download_stream
```

That would result in `downloader` and `downloader_stream` objects created at different times. Is this a problem?